### DISCUSSION TITLE
How fast can you bring python to blank windows?

### DISCUSSION THREAD

**Discussion by liquidcarbon** [winpython/winpython#1342] [Show and tell]
Hello!  I've used winpython in the past, thank you for your work!

I'm curious if you'd be open to creating or offering existing benchmarks on how long it takes to install python on Windows.

Here's a tool I made: https://github.com/liquidcarbon/puppy

And this is how you use it.  The video is real-time, but the first install is probably slower than what's depicted, because it's likely using cached resources.

https://private-user-images.githubusercontent.com/47034358/330357477-16821b3b-f049-4e0a-873b-f90fcdc3f5a2.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3NDQ5MzQsIm5iZiI6MTcxNTc0NDYzNCwicGF0aCI6Ii80NzAzNDM1OC8zMzAzNTc0NzctMTY4MjFiM2ItZjA0OS00ZTBhLTg3M2ItZjkwZmNkYzNmNWEyLm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE1VDAzNDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4YmZmNjRiMWQ2Y2MyYmExMjlmN2IzN2UzNGY1NjU1MTk0ZWY0OTgwMGIyYzZkNWE1Zjc2MGQ4YjMxOWY4MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IULQjRben34Vpx-ixkS-LKoqu5H3xe5Zr9pbuRvGTt8

**Comment by stonebig** [comment]
There is no competition on installing / un-7zipping Winpython.
Effort to accelerate it would require to switch to zstd when it will be officially supported  in Windows, and but Then the archive size will be 20% bigger

**Comment by stonebig** [comment]
Otherwise, you would need to use uv and hashes of the packages to download .... that is the best option, when uv will stop versionning every week and disturb antivirusses

**Comment by stonebig** [comment]
To make WinPython fast to install, we would need either:
- Windows having zstd in it's .7z integration
- a speedier and "exact hash" pip installer
